plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java

```diff
@@ -312,12 +317,38 @@ public static String quotedTableName(SchemaTableName name)
 
     private static String quotedName(String name)
     {
-        if (SIMPLE_NAME.matcher(name).matches()) {
+        if (UNQUOTED_IDENTIFIER_PATTERN.matcher(name).matches()) {
             return name;
         }
         return '"' + name.replace("\"", "\"\"") + '"';
     }
 
+    public static String fromColumnToIdentifier(String column)
+    {
+        if (UNQUOTED_IDENTIFIER_PATTERN.matcher(column).matches()) {
+            return column;
+        }
+        return quotedName(column);
+    }
+
+    public static String fromIdentifierToColumn(String identifier)
```

Comment:
The method introduced here is unused (in the commit which introduced it), and it's unclear what's the context in which is should be used.
Squash the changes with the next commit.

Reply:
Later it becomes used in `PartitionFields` which is the context in which it's comprehensible. 
(otherwise the "identifier" will be misleading, as normally `String` identifier should not be expected to have any quotes inside (or the quotes be treated literal).

Move the method to `PartitionFields`

Reply:
The placement in `IcebergUtils` had been discussed in #12872, the need to parse quoted identifiers also exist in other table properties (sort_order, orc_bloom_filter, ...).